packages/ag-charts-enterprise/src/series/treemap/treemapSeries.ts

```diff
@@ -360,14 +361,43 @@ export class TreemapSeries extends _ModuleSupport.HierarchySeries<
         let style = getShapeStyle(baseStyle, fillGradientDefaults, fillPatternDefaults, fillImageDefaults);
 
         if (itemStyler != null && datumIndex != null) {
+            const datumIndexString = datumIndex.join(':');
             const overrides = this.cachedDatumCallback(
                 createDatumId(datumIndex.join(':'), isHighlight ? 'highlight' : 'node'),
                 () => {
+                    const activeHighlight = ctx.highlightManager?.getActiveHighlight();
+                    let highlightState: HighlightState;
+
+                    if (isHighlight) {
+                        // This node is highlighted
+                        if (
+                            datumIndex &&
+                            activeHighlight &&
+                            'datumIndex' in activeHighlight &&
+                            Array.isArray(activeHighlight.datumIndex) &&
+                            activeHighlight.datumIndex?.join(':') === datumIndexString
+                        ) {
+                            highlightState = 'highlighted-item';
+                        } else {
+                            highlightState = 'highlighted-series';
+                        }
+                    } else if (activeHighlight) {
+                        // Something else is highlighted
+                        if (activeHighlight.series === this) {
+                            highlightState = 'unhighlighted-item';
+                        } else {
+                            highlightState = 'unhighlighted-series';
+                        }
+                    } else {
+                        highlightState = 'none';
+                    }
```

Comment:
can we use the Series.getHighlightState function for this? I think we would want the highlightState to be correct maybe the `isItemHighlighted` function needs to be overridden for treemap